test/e2e/certificates_test.go

```diff
@@ -0,0 +1,112 @@
+	// loop all Windows nodes
+	for _, winNode := range gc.allNodes() {
+		// spin-off subtests to ensure the CA bundle file is verified in all Windows nodes
+		t.Run("node/"+winNode.Name, func(t *testing.T) {
+			err := tc.waitForKubeletCACertificateInNode(&winNode)
+			assert.NoErrorf(t, err, "kubelet CA certificate should be present in node %S", winNode.Name)
+		})
+	}
+}
+
+// forceKubeApiServerCertificateRotation forces the certificate rotation by setting the `certificate-not-after`
+// annotation to nil in the `kube-apiserver-to-kubelet-signer` secret
+func (tc *testContext) forceKubeApiServerCertificateRotation() error {
+	// note that the slash had to be encoded as ~1 because it's a reference: https://www.rfc-editor.org/rfc/rfc6901#section-3
+	patchData := fmt.Sprint(`[{"op":"replace","path":"/metadata/annotations/auth.openshift.io~1certificate-not-after","value": null }]`)
+	// invoke request
+	_, err := tc.client.K8s.CoreV1().Secrets("openshift-kube-apiserver-operator").Patch(context.TODO(),
+		"kube-apiserver-to-kubelet-signer", types.JSONPatchType, []byte(patchData), meta.PatchOptions{})
+	return err
+}
+
+// waitForKubeletCACertificateInNode waits for the kubelet CA certificate to be present in the CA bundle file for the
+// given Windows node. After retry.Interval runs an SSH job to fetch the CA bundle, if the job fails or
+// kubelet CA is not present in the node, retries every retry.Interval until retry.Timeout is reached.
+func (tc *testContext) waitForKubeletCACertificateInNode(node *core.Node) error {
+	addr, err := controllers.GetAddress(node.Status.Addresses)
+	if err != nil {
+		return err
+	}
+	// CA bundle location in Windows node. i.e. "C:\k\kubelet-ca.crt"
+	caBundlePath := windows.GetK8sDir() + nodeconfig.KubeletClientCAFilename
+	// PowerShell command to fetch content in the file
+	command := fmt.Sprintf("Get-Content -Raw -Path %s", caBundlePath)
+	// wait retry.Interval and verify the CA bundle content, try if needed
+	return wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
+		// invoke command
+		bundleContent, err := tc.runPowerShellSSHJob("kubelet-ca-bundle-content", command, addr)
+		if err != nil {
+			// retry
+			log.Printf("error fetching CA bundle in node %s with address %s, retrying in %s. %v",
+				node.Name, addr, retry.Interval, err)
+			return false, nil
+		}
+		// get the ConfigMap that contains the kubelet client CA
+		cm, err := tc.client.K8s.CoreV1().ConfigMaps(certificates.KubeApiServerOperatorNamespace).Get(context.TODO(),
+			certificates.KubeAPIServerServingCAConfigMapName, meta.GetOptions{})
+		if err != nil {
+			return false, errors.Wrap(err, "error getting kubelet client CA ConfigMap")
+		}
+		// parse bundle from ConfigMap
+		kubeletCABytes, err := certificates.GetCAsFromConfigMap(cm, certificates.CABundleKey)
+		if err != nil {
+			return false, errors.Wrap(err, "error parsing CA bundle from ConfigMap")
+		}
+		kubeletCAString := strings.TrimSpace(string(kubeletCABytes))
```

Comment:
It's a trade-off, to fetch the kubelet CA ConfigMap first and pass it along, we need to wait for the kube-apiserver-operator to update the ConfigMap with the new certificate (the one resulting from the rotation with the longest validity). I do have an "estimation" for that, about 5 seconds. It may and will vary depending on many factors, cluster's overload, etc.

To reduce any possible [flakes](https://github.com/openshift/windows-machine-config-operator/pull/1013#discussion_r854534019), I decided to move this code here. Yes, It adds an extra API call per Windows node but is worth it compared to relying on reference data that may not be the latest.